internal/systemd/systemd_test.go

```diff
@@ -5,80 +5,253 @@ import (
-				}
-			} else {
-				if err != nil {
-					t.Fatal(err)
-				}
-				got, err := conn.GetUnitState(filepath.Base(test.input.unitFile))
-				if err != nil {
-					t.Fatal(err)
-				}
-				if !cmp.Equal(got, test.want) {
-					t.Errorf("%v", cmp.Diff(got, test.want))
-				}
+			if conn == nil {
+				t.Error("expected non-nil connection")
+				return
 			}
+
+			conn.Close()
 		})
 	}
 }
+
+func TestReload(t *testing.T) {
+	if _, has := os.LookupEnv("DBUS_SESSION_BUS_ADDRESS"); !has {
+		t.Skip("DBUS_SESSION_BUS_ADDRESS undefined")
+	}
+
+	conn, err := NewConnectionContext(context.Background(), ConnectionTypeUser)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer conn.Close()
+
+	err = conn.Reload()
+	if err != nil {
+		t.Logf("Reload failed: %v", err)
+	}
+}
+
+func TestUnitOperationsValid(t *testing.T) {
+	if _, has := os.LookupEnv("DBUS_SESSION_BUS_ADDRESS"); !has {
+		t.Skip("DBUS_SESSION_BUS_ADDRESS undefined")
+	}
+
+	conn, err := NewConnectionContext(context.Background(), ConnectionTypeUser)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer conn.Close()
+
+	unitFile := "testdata/simple.service"
+	abs, err := filepath.Abs(unitFile)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = conn.conn.LinkUnitFilesContext(conn.ctx, []string{abs}, true, false)
+	if err != nil {
+		t.Fatal(err)
+	}
+	unitName := filepath.Base(unitFile)
+
+	t.Run("enable_unit", func(t *testing.T) {
+		err := conn.EnableUnit(unitName, false, true)
+		if err != nil {
+			t.Fatalf("EnableUnit failed: %v", err)
+		}
+	})
+
+	t.Run("get_unit_state", func(t *testing.T) {
+		state, err := conn.GetUnitState(unitName)
+		if err != nil {
+			t.Fatalf("GetUnitState failed: %v", err)
+		}
+		if state == "" {
+			t.Error("expected non-empty state")
+		}
+	})
+
+	t.Run("start_unit", func(t *testing.T) {
+		err := conn.StartUnit(unitName, true)
+		if err != nil {
+			t.Fatalf("StartUnit failed: %v", err)
+		}
+
+		state, err := conn.GetUnitState(unitName)
+		if err != nil {
+			t.Fatalf("GetUnitState failed: %v", err)
+		}
+		if state != "active" {
+			t.Errorf("expected unit to be active, got %q", state)
+		}
+	})
+
+	t.Run("stop_unit", func(t *testing.T) {
+		err := conn.StopUnit(unitName, true)
+		if err != nil {
+			t.Fatalf("StopUnit failed: %v", err)
+		}
+
+		state, err := conn.GetUnitState(unitName)
+		if err != nil {
+			t.Fatalf("GetUnitState failed: %v", err)
+		}
+		if state == "active" {
+			t.Errorf("expected unit to be inactive after stop, got %q", state)
+		}
+	})
+
+	t.Run("disable_unit", func(t *testing.T) {
+		if err = conn.DisableUnit(unitName, false, true); err != nil {
+			t.Fatalf("DisableUnit failed: %v", err)
+		}
+	})
+}
+
+func TestUnitOperationsInvalid(t *testing.T) {
+	if _, has := os.LookupEnv("DBUS_SESSION_BUS_ADDRESS"); !has {
+		t.Skip("DBUS_SESSION_BUS_ADDRESS undefined")
+	}
+
+	conn, err := NewConnectionContext(context.Background(), ConnectionTypeUser)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer conn.Close()
```

Comment:
Please use closure for `conn.Close()`, because this function call can return `error`. It would be good to call `t.Errorf()` in such case or handle such situation somehow.

Reply:
Oh, I didn't realize that this function does not return error at all. Strange.